### ISSUE TITLE
WPF - Crash after docking into another panel

### ISSUE THREAD

**Issue by bsx1990** [cefsharp/CefSharp#3076]
Hi, Cefsharp
  I tried to dock panel Message3 to Message2, it successed, but after that, I found the web page, which embeded with cefsharp, crashed.
  But Message1 works very well, no matter drag it out and dock it again.
  Could you help to have a look at the demo?

- **What version of the product are you using?**
    - 75.1.142

- **What architecture x86 or x64?**
    - x64

- **On what operating system?**
    - Win7

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
    - WPF

**Comment by amaitland** [comment]
> Please only create an issue if you can reproduce the problem with version 79.1.360 or greater.

You'll need to test with `79.1.360`, you are using an old unsupported version.

> Please no binary attachments (zip, 7z, etc), code needs to be easily reviewed in a web browser.

As per the [Bug Report Template](https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/bug_report.md), no `zip` files, thank you.

> If code is required to reproduce your problem then please provide one of the following
Fork the MinimalExample and push your changes to GitHub (this is the preferred option).
Use a code sharing service list Gist or Pastebin

If you can reproduce the problem with version `79.1.360` the please fork the `MinimalExample` and provide an example, squash all your changes into a single commit.

**Comment by amaitland** [comment]
I suspect this is a duplicate of #2824 for which you'll need to upgrade.

**Comment by bsx1990** [comment]
Already create a pull request for this issue

**Comment by bsx1990** [comment]
> I suspect this is a duplicate of #2824 for which you'll need to upgrade.



> > Please only create an issue if you can reproduce the problem with version 79.1.360 or greater.
> 
> You'll need to test with `79.1.360`, you are using an old unsupported version.
> 
> > Please no binary attachments (zip, 7z, etc), code needs to be easily reviewed in a web browser.
> 
> As per the [Bug Report Template](https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/bug_report.md), no `zip` files, thank you.
> 
> > If code is required to reproduce your problem then please provide one of the following
> > Fork the MinimalExample and push your changes to GitHub (this is the preferred option).
> > Use a code sharing service list Gist or Pastebin
> 
> If you can reproduce the problem with version `79.1.360` the please fork the `MinimalExample` and provide an example, squash all your changes into a single commit.

I am not sure if I should create a pull request for the demo. Please have a look at it.

**Comment by amaitland** [comment]
> I am not sure if I should create a pull request for the demo. Please have a look at it.

Just a link to the branch is sufficient thanks.

https://github.com/bsx1990/CefSharp.MinimalExample/tree/cefsharp/79

**Comment by bsx1990** [comment]
> > I am not sure if I should create a pull request for the demo. Please have a look at it.
> 
> Just a link to the branch is sufficient thanks.
> 
> https://github.com/bsx1990/CefSharp.MinimalExample/tree/cefsharp/79

So could anyone help to look at this issue?
Thanks!

**Comment by amaitland** [comment]
Please provide a stack trace, if the exception comes from `libcef.dll` then follow the instructions at https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#loading-native-symbols-for-easier-diagnosis

**Comment by bsx1990** [comment]
I have downloaded the pub file of libcef.dll, but I am wrong.
It doesn't have any stack trace, the program just hangs.

**Comment by amaitland** [comment]
As yet you haven't provided a complete bug report, please complete the following two sections with as much detail as possible.

**What steps will reproduce the problem?**

**What is the expected output? What do you see instead?**

What hangs exactly? Please provide a detailed description.

**Comment by bsx1990** [comment]
> As yet you haven't provided a complete bug report, please complete the following two sections with as much detail as possible.
> 
> **What steps will reproduce the problem?**
> 
> **What is the expected output? What do you see instead?**
> 
> What hangs exactly? Please provide a detailed description.

I have already provide a demo, could you run and see it?
Here are the steps:
1. Run the application
2. Click Reassign button after the webpage loaded
3. Now you can see a popup window, now do anything to the web page, you can find that the web page had hanged, no response.

**Comment by bsx1990** [comment]
I'm hard to say what is the expected output. Because I just want a normal page, but it could not response after step 2.

**Comment by amaitland** [comment]
Try comment our the `Close()` call, the `Browser` is probably being `Disposed` when the original `Window` is closed. If that's the case then it's pretty easy to workaround.

**Comment by bsx1990** [comment]
> Try comment our the `Close()` call, the `Browser` is probably being `Disposed` when the original `Window` is closed. If that's the case then it's pretty easy to workaround.

Comment 'Close()' works fine. But my actual production code is just similar with 'Close()', not 'Close()'

I use DevExpress as a UI component library. And I had provided a zip at the first post. I'm not sure if it disposed the browser when docking to another panel or not, can you help to see the first zip application which I provided at the first message?

**Comment by amaitland** [comment]
You can assign the CleanupElement property to your new window before calling close on the previous parent window. 

http://cefsharp.github.io/api/79.1.x/html/P_CefSharp_Wpf_ChromiumWebBrowser_CleanupElement.htm

No zip files, please stop asking.
